### DISCUSSION TITLE
libusb class code how to choose about usb3.0

### DISCUSSION THREAD

**Discussion by zhanggaojian** [libusb/libusb#1330] [Q&A] [answered]
Hi，
what is the libusb class code when use usb3.0 to communicate
thanks.

**Comment by mcuee** [comment]
What do you mean by that? libusb does not decide the USB connection speed. It is up to the OS and your device.

**Reply by diabolo38** [reply, marked as answer]
that i know usb speed is not linked in anyway to class code 
this is discovered by os and app   using device and others descriptor  (bos , ep compagin etc ...) 
an easy to find working speed (maybe unperfect) it to look at a bulk end point from their the max packet size 
shall be 64 for USB 1 or USB 2  FS , 512 for USB2 HS and 1024 for USB 3.x SS

**Reply by zhanggaojian** [reply]
thank you very much!
